[extractor/tiktok] Fix TikTokIE

bashonly opened the pull request:
</details>

<!--

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible

-->

Fixes video extraction via API. Updated app versions and removed broken `aweme/detail` endpoint request.

<details open><summary>Template</summary> <!-- OPEN is intentional -->

<!--

# PLEASE FOLLOW THE GUIDE BELOW

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes `[ ]` relevant to your *pull request* (like [x])
- Use *Preview* tab to see how your *pull request* will actually look like

-->

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

pukkandan commented on yt_dlp/extractor/tiktok.py:
```diff
@@ -46,7 +46,8 @@ def _get_sigi_state(self, webpage, display_id):
             return
-        self._request_webpage(HEADRequest('https://www.tiktok.com'), None, note='Setting up session', fatal=False)
+        self._request_webpage(HEADRequest('https://www.tiktok.com'), None,
+                              headers={'User-Agent': 'User-Agent:Mozilla/5.0'}, note='Setting up session', fatal=False)
```
```suggestion
                              headers={'User-Agent': 'Mozilla/5.0'}, note='Setting up session', fatal=False)
```
typo?

bashonly replied:
this is deliberate; seemingly any HTTP/1.1 request to `tiktok.com` will now timeout unless the UA is exactly `User-Agent:Mozilla/5.0` (see also line 556)

although someziggyman looks to be having an issue even with this UA, hmm

someziggyman commented:
thanks for taking a look at it. Run a quick test with this link:
https://www.tiktok.com/@manoloteachesgolf/video/7141068332156898606?is_copy_url=1&is_from_webapp=v1

It seems to work fine, however take some time to fetch data "setting up session" step, to be exact. Then it times out with a warning: WARNING: [TikTok] Unable to download webpage: The read operation timed out

But extraction is successful. Please see the log below:

./yt-dlp -v -F "https://www.tiktok.com/@manoloteachesgolf/video/7141068332156898606?is_copy_url=1&is_from_webapp=v1"
[debug] Command-line config: ['-v', '-F', 'https://www.tiktok.com/@manoloteachesgolf/video/7141068332156898606?is_copy_url=1&is_from_webapp=v1']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.09.01 [5d7c7d656] (zip)
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python 3.10.6 (CPython 64bit) - macOS-12.6-arm64-arm-64bit 
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: phantomjs 2.1.1
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1716 extractors
**[TikTok] Setting up session
WARNING: [TikTok] Unable to download webpage: The read operation timed out**
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@manoloteachesgolf/video/7141068332156898606?is_copy_url=1&is_from_webapp=v1
[TikTok] 7141068332156898606: Downloading video details
[debug] Sort order given by extractor: quality, codec, size, br
[debug] Formats sorted by: hasvid, ie_pref, quality, vcodec, acodec, filesize, fs_approx, tbr, vbr, abr, lang, res, fps, hdr:12(7), channels, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 7141068332156898606:
ID                     EXT RESOLUTION │  FILESIZE   TBR PROTO │ VCODEC   VBR ACODEC ABR MORE INFO
─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
download_addr-0        mp4 576x1024   │  27.68MiB       https │ h264         aac        Download video, watermarked (API)
download_addr-1        mp4 576x1024   │  27.68MiB       https │ h264         aac
.... and so on .....

bashonly commented:
I removed the TikTokBaseIE `_real_initialize` function, as I don't believe it serves a purpose anymore:

Seemingly any HTTP/1.1 request to `tiktok.com` will timeout. There is a workaround of using the strange UA string of `User-Agent:Mozilla/5.0` [sic], but this does not seem to work for all users, as evidenced by someziggyman's testing.

I assume the purpose of the `_real_initialize` was to get cookies before requesting video/profile webpages, since the API requests do not use the cookies (different domain). But if the `_real_initialize` request times out, then any subsequent webpage requests will timeout also, and cookies are pointless. And, if the UA workaround actually works (like it does for me), then webpage extraction succeeds even without cookies.


Also removed some code duplication that I had initially missed. Should be good to go now.

pukkandan commented on yt_dlp/extractor/tiktok.py:
```diff
@@ -467,6 +460,7 @@ class TikTokIE(TikTokBaseIE):
             'uploader_id': '7036055384943690754',
             'uploader_url': 'https://www.tiktok.com/@MS4wLjABAAAATh8Vewkn0LYM7Fo03iec3qKdeCUOcBIouRk1mkiag6h3o_pQu_dUXvZ2EZlGST7_',
             'track': 'Promoted Music',
+            'artist': '',
```
This appears to be a bug in the code. Adding a `or None` where this extraction is happening should fix it

bashonly replied:
fixed

fireattack commented:
This version (`master`) still gives the warning "Unable to find video in feed; please report this issue on...", is it expected? Video is actually downloaded (but it's the watermarked version).

```
G:\_temp\yt-dlp>python -m yt_dlp "https://www.tiktok.com/@animax_jp/video/7146096066105019649" --ignore-config --verbose
[debug] Command-line config: ['https://www.tiktok.com/@animax_jp/video/7146096066105019649', '--ignore-config', '--verbose']
[debug] Encodings: locale cp936, fs utf-8, pref cp936, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version 2022.09.01 [5d7c7d656] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 573a98d6f
[debug] Python 3.8.7 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 2022-08-29-git-f99d15cca0-full_build-www.gyan.dev (setts), ffprobe 2022-08-29-git-f99d15cca0-full_build-www.gyan.dev, rtmpdump 2.3
[debug] Optional libraries: Cryptodome-3.9.4, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, secretstorage-3.3.1, sqlite3-2.6.0, websockets-10.1
[debug] Proxy map: {}
[debug] Loaded 1681 extractors
[debug] [TikTok] Extracting URL: https://www.tiktok.com/@animax_jp/video/7146096066105019649
[TikTok] 7146096066105019649: Downloading video feed
WARNING: [TikTok] 7146096066105019649: Unable to find video in feed; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U; trying with webpage
[TikTok] 7146096066105019649: Downloading webpage
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 7146096066105019649: Downloading 1 format(s): 0
[debug] Invoking http downloader on "https://v16m-webapp.tiktokcdn-us.com/2100072f4134921596319a6f152fb519/63399baa/video/tos/alisg/tos-alisg-pve-0037/7d240ee7c9504bcb802c588b147d52c2/?a=1988&ch=0&cr=0&dr=0&lr=tiktok_m&cd=0%7C0%7C1%7C0&cv=1&br=3058&bt=1529&cs=0&ds=3&ft=ebtHKH-qMyq8Z.N6Dhe2Nft-fl7Gb&mime_type=video_mp4&qs=0&rc=aTxoaGg7ODtlOTQ5OzVoZUBpajk3amc6ZjUzZjMzODgzNEAzMzFjYTIvNWExNl4xYTYuYSNrcS5rcjQwZWpgLS1kLy1zcw%3D%3D&l=20221002080935D7C801CA2595C0372049"
[debug] File locking is not supported. Proceeding without locking
[download] Destination: #誰でも二次元 #animax #animaxmusix #アニメミュージックスターオーディション  #伊藤美来  [7146096066105019649].mp4
[download] 100% of 2.13MiB in    00:00 at 12.09MiB/s
```

bashonly commented:
@fireattack some videos are unavailable via the `feed` API endpoint. This was known to happen with videos where the audio track had been taken down (tracked in issue #4992), but your case seems to be different since the video is still available via webpage. Hmm
